cpuminer.go

```diff
@@ -251,15 +251,17 @@ func (m *CPUMiner) solveBlock(msgBlock *wire.MsgBlock, ticker *time.Ticker, quit
 	// Note that the entire extra nonce range is iterated and the offset is
 	// added relying on the fact that overflow will wrap around 0 as
 	// provided by the Go spec.
-	for extraNonce := uint64(0); extraNonce < maxExtraNonce; extraNonce++ {
```

Comment:
After looking at this a bit more closely, while my initial comments are also true in regards to the fact that the difficulties are so low for the CPU miner that it's only ever been used with blocks that have solutions in the 32-bit nonce space,
the wrap around behavior on the extra nonce is actually desired here as the comment notes.  The reason is so that the mining process will continue forever until a solution is found.

Looking further in the code, we can see that the block header timestamp is periodically updated during the iteration of the regular nonce space which will ultimately change the hash and means any extra nonce values that were previously tested may now have become valid, and thus wrapping around is desirable so that eventually a solution will be found.

However, in the case of regular nonce space, while the same applies in terms of the timestamp being updated and hence the wrap around behavior would ultimately find a solution there too, wrapping around at that point would mean the extra nonce would never be updated.

In practice, the entire function exits early when there is an updated template anyway (see the `case <-ticker.C` code, so none of this is going to currently be hit, but I do think the code should be correct regardless.

Given all of that, what do you think about just removing the condition from the extra nonce for loop to help make it clear it is intentionally infinite and tests the entire space, and then, in the case of the regular nonce space, retaining the for loop to set the initial condition and increment the counter of the nonce,  but moving the break condition to the end instead?  I think that might make it more clear:

e.g:

```Go
	// Note that the entire extra nonce range is iterated and the offset is
	// added relying on the fact that overflow will wrap around 0 as provided by
	// the Go spec.
	for extraNonce := uint64(0); ; extraNonce++ {
		// Update the extra nonce in the block template header with the new
		// value.
		littleEndian.PutUint64(header.ExtraData[:], extraNonce+enOffset)


		// Search through the entire nonce range for a solution while
		// periodically checking for early quit and stale block conditions along
		// with updates to the speed monitor.
		//
		// Note that the break condition is intentionally not checked until the
		// end of the loop to ensure the entire nonce space is explored.
		for nonce := uint32(0); ; nonce++ {
			// code....
		

			if nonce == maxNonce {
				break
			}
		}
	}
```

Reply:
Thanks for the explanation!  Regarding the existing comment about overflow, I had interpreted that only to mean that adding any offset to the current extraNonce value would yield the desired behavior, and not that the loop itself was desired to run indefinitely.  I added another sentence to make this more explicit.

I also like your for loop construction better (I'm new to Golang, and didn't think to omit the condition expression), so I used it in the updated code.

Finally, I took the time to wordsmith a bit more commentary around these two loops and why they differ slightly.  Hopefully it's clearer now for the next developer who comes along.

Reply:
Looks good.  The comments help make it more clear as well.  Thanks for the updates.